### ISSUE TITLE
Amazon Linux EC2 instances experience a "Killed" condition when the script runs

### ISSUE THREAD

**Issue by marknca** [deep-security/aws-waf#7]
Output from a standard run (that succeeds locally):

```
Reading AWS credentials from /home/ec2-user/.aws/credentials
Caching TBUIDS for rule matching
Caching patterns for rule matching
Attempting to connect to Deep Security at app.deepsecurity.trendmicro.com:443
Connected to the Deep Security Manager at app.deepsecurity.trendmicro.com
Connected to AWS waf
Connected to AWS ec2
Connected to AWS elb
Connected to AWS cloudfront
Requesting information from Deep Security about your deployment
Requesting rules from the Deep Security manager. This will take a few seconds...
Killed
```

**Comment by marknca** [comment]
Traced the issue to the SOAP call for ```integrityRuleRetrieveAll```. There appears to be an issue in the manner that the SUDS library handles this call (it typically returns the largest set of information and contains unicode characters, not sure which is the issue).

Testing with another SOAP library ([osa](https://bitbucket.org/sboz/osa/wiki/Home)) does not have this issue. Anecdotally, performance seems significantly better with osa as well.

**Comment by marknca** [comment]
Started a new branch to implement osa as the SOAP library in lieu of suds.

[branch/swap-soap-module](https://github.com/deep-security/aws-waf/tree/swap-soap-module)

**Comment by marknca** [comment]
Removed 'swap-soap-module' branch on this repo and moved the code change to the core [deep-security-py](https://github.com/deep-security/deep-security-py/tree/new-soap-library) repo where it should be. Once complete, I'll open a new branch this repo to implement the updates deep-security-py module.

**Comment by marknca** [comment]
_Short story_: I have a tentative fix up in a new branch on this repo. Checkout [https://github.com/deep-security/aws-waf/tree/new-sdk](https://github.com/deep-security/aws-waf/tree/new-sdk).

This fix includes the yet-to-be-release v2 of the deepsecurity-py project. The functionality for the AWS WAF script ```ds-to-aws-waf.py``` remains **unchanged**.

_Long story_: v2 of the deepsecurity-py project uses the [suds](https://fedorahosted.org/suds/) module to access Deep Security's SOAP API. While a bit slow, the module generally works great. In Deep Security's case, the module works perfectly _except_ when calling ```integrityRuleRetrieveAll``` from an Amazon Linux EC2 instance.

After much debugging, hair pulling, and frustration, I traced the issue back to how the results of that call (a ~4000 node document including unicode characters) are processed by the python stdlib (expat specifically). This particular response causes the expat process to violate a number of conditions resulting in the python interpreter (and any associated debuggers!) being *killed* by the system.

After attempting to use a number of other SOAP modules, a v2 of the deepsecurity-py project was started using a customer SOAP interface (yes, that should cause shudders but it turns out it was not only faster but easier to debug as it reduced the code path from several hundred lines to around 50 for each SOAP call).

I've merged that new v2 into the ```new-idk``` branch of this repo and all tests (including on Amazon Linux EC2 instances) so far have been successful.
